applications/convection_diffusion_application/python_scripts/convection_diffusion_base_solver.py

```diff
@@ -0,0 +1,757 @@
+            if (self.settings["restart_settings"]["save_restart"].GetBool() == True):
+                # the check if this step is a restart-output step is done internally
+                self.get_restart_utility().SaveRestart()
+
+    def Solve(self):
+        if self.settings["clear_storage"].GetBool():
+            self.Clear()
+        mechanical_solution_strategy = self.get_mechanical_solution_strategy()
+        mechanical_solution_strategy.Solve()
+
+    def InitializeSolutionStep(self):
+        self.get_mechanical_solution_strategy().InitializeSolutionStep()
+
+    def Predict(self):
+        self.get_mechanical_solution_strategy().Predict()
+
+    def SolveSolutionStep(self):
+        is_converged = self.get_mechanical_solution_strategy().SolveSolutionStep()
+        return is_converged
+
+    def FinalizeSolutionStep(self):
+        self.get_mechanical_solution_strategy().FinalizeSolutionStep()
+
+    def AdvanceInTime(self, current_time):
+        dt = self.ComputeDeltaTime()
+        new_time = current_time + dt
+        self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+        self.main_model_part.CloneTimeStep(new_time)
+
+        return new_time
+
+    def ComputeDeltaTime(self):
+        return self.settings["time_stepping"]["time_step"].GetDouble()
+
+    def GetComputingModelPart(self):
+        return self.main_model_part.GetSubModelPart(self.settings["computing_model_part_name"].GetString())
+
+    def ExportModelPart(self):
+        name_out_file = self.settings["model_import_settings"]["input_filename"].GetString()+".out"
+        file = open(name_out_file + ".mdpa","w")
+        file.close()
+        KratosMultiphysics.ModelPartIO(name_out_file, KratosMultiphysics.IO.WRITE).WriteModelPart(self.main_model_part)
+
+    def SetEchoLevel(self, level):
+        self.get_mechanical_solution_strategy().SetEchoLevel(level)
+
+    def Clear(self):
+        self.get_mechanical_solution_strategy().Clear()
+
+    def Check(self):
+        self.get_mechanical_solution_strategy().Check()
+
+    #### Specific internal functions ####
+
+    def get_solution_scheme(self):
+        if not hasattr(self, '_solution_scheme'):
+            self._solution_scheme = self._create_solution_scheme()
+        return self._solution_scheme
+
+    def get_convergence_criterion(self):
+        if not hasattr(self, '_convergence_criterion'):
+            self._convergence_criterion = self._create_convergence_criterion()
+        return self._convergence_criterion
+
+    def get_linear_solver(self):
+        if not hasattr(self, '_linear_solver'):
+            self._linear_solver = self._create_linear_solver()
+        return self._linear_solver
+
+    def get_builder_and_solver(self):
+        if not hasattr(self, '_builder_and_solver'):
+            self._builder_and_solver = self._create_builder_and_solver()
+        return self._builder_and_solver
+
+    def get_mechanical_solution_strategy(self):
```

Comment:
why a mechanical solution strategy in a convection_diffusion one?

Reply:
CP from structural, sorry